### ISSUE TITLE
Dataset annotation

### ISSUE THREAD

**Issue by xwhkkk** [yisun98/SOLC#1]
Hello, thanks for sharing your work.
What's the annotation of Dataset WHU-OPT-SAR ? I noticed a folder named **lib**, what's that mean ?
Thanks!

**Comment by yisun98** [comment]
Thanks for your attention. Please refer to "https://github.com/AmberHen/WHU-OPT-SAR-dataset" for details. This link provides the source data.  1. The original paper “https://doi.org/10.1016/j.jag.2021.102638” provides the annotation as follows. 0->background，10->farmland，20->city，30->village，40->water，50->forest，60->road，70->others.  2. lib or lbl? Sorry, we did not find the folder named **lib**.  The folder named lbl is the annotation of Dataset WHU-OPT-SAR.

**Comment by bradhuang123** [comment]
Hello, there! It seems that the instruction of your work is not detailed enough for me to run your code. For instance, the pip install requirements.txt did not work. There is no such file! I hope I can hopefully get a more detailed instrction on your project, thank you very much.

**Comment by shelp12138** [comment]
你好，我看到你们实现的mcanet的编码器只用了一层卷积，而不是原文中的resnet101，你们最终的跑出来的结果是基于一层卷积的编码器还是基于resnet101的呢？

**Comment by 12cyan** [comment]
lbl文件夹 都是tiff图片，具体的标签信息在哪里呢

**Comment by yisun98** [comment]
> 你好，我看到你们实现的mcanet的编码器只用了一层卷积，而不是原文中的resnet101，你们最终的跑出来的结果是基于一层卷积的编码器还是基于resnet101的呢？

我暂未发现一层卷积部分：
        self.sar_en1 = _EncoderBlock(1, 64) # 256->128, 1->64
        self.sar_en2 = _EncoderBlock(64, 256)  # 128->64, 64->256
        self.sar_en3 = _EncoderBlock(256, 512)  # 64->32, 256->512
        self.sar_en4 = _EncoderBlock(512, 1024, downsample=False)  # 32->32 *** , 512->1024
        self.sar_en5 = _EncoderBlock(1024, 2048, downsample=False)  # 32->32 *** , 1024->2048

        self.opt_en1 = _EncoderBlock(4, 64) # 256->128, 4->64
        self.opt_en2 = _EncoderBlock(64, 256)  # 128->64, 64->256
        self.opt_en3 = _EncoderBlock(256, 512)  # 64->32, 256->512
        self.opt_en4 = _EncoderBlock(512, 1024, downsample=False)  # 32->32 *** , 512->1024
        self.opt_en5 = _EncoderBlock(1024, 2048, downsample=False)  # 32->32 *** , 1024->2048

**Comment by yisun98** [comment]
> lbl文件夹 都是tiff图片，具体的标签信息在哪里呢

具体的标签信息保存在lbl文件夹中，标签的保存格式为tif图像，可读取tif图像查看每个像素对应的标签。

**Comment by shelp12138** [comment]
> > 你好，我看到你们实现的mcanet的编码器只用了一层卷积，而不是原文中的resnet101，你们最终的跑出来的结果是基于一层卷积的编码器还是基于resnet101的呢？
> 
> 我暂未发现一层卷积部分： self.sar_en1 = _EncoderBlock(1, 64) # 256->128, 1->64 self.sar_en2 = _EncoderBlock(64, 256) # 128->64, 64->256 self.sar_en3 = _EncoderBlock(256, 512) # 64->32, 256->512 self.sar_en4 = _EncoderBlock(512, 1024, downsample=False) # 32->32 *** , 512->1024 self.sar_en5 = _EncoderBlock(1024, 2048, downsample=False) # 32->32 *** , 1024->2048
> 
> ```
>     self.opt_en1 = _EncoderBlock(4, 64) # 256->128, 4->64
>     self.opt_en2 = _EncoderBlock(64, 256)  # 128->64, 64->256
>     self.opt_en3 = _EncoderBlock(256, 512)  # 64->32, 256->512
>     self.opt_en4 = _EncoderBlock(512, 1024, downsample=False)  # 32->32 *** , 512->1024
>     self.opt_en5 = _EncoderBlock(1024, 2048, downsample=False)  # 32->32 *** , 1024->2048
> ```
class _EncoderBlock(nn.Module):
    def __init__(self, in_channels, out_channels, downsample=True):
        super(_EncoderBlock, self).__init__()
        layers = [
            nn.Conv2d(in_channels, out_channels, kernel_size=3, padding=1),
            nn.BatchNorm2d(out_channels),
            nn.ReLU(inplace=True),
        ]
        if downsample:
            layers.append(nn.MaxPool2d(kernel_size=2, stride=2))

        self.encode = nn.Sequential(*layers)

    def forward(self, x):
        return self.encode(x)
这个 _EncoderBlock确实只包含一个conv-BN-ReLU块

**Comment by yisun98** [comment]
> > > 你好，我看到你们实现的mcanet的编码器只用了一层卷积，而不是原文中的resnet101，你们最终的跑出来的结果是基于一层卷积的编码器还是基于resnet101的呢？
> > 
> > 
> > 我暂未发现一层卷积部分： self.sar_en1 = _EncoderBlock(1, 64) # 256->128, 1->64 self.sar_en2 = _EncoderBlock(64, 256) # 128->64, 64->256 self.sar_en3 = _EncoderBlock(256, 512) # 64->32, 256->512 self.sar_en4 = _EncoderBlock(512, 1024, downsample=False) # 32->32 *** , 512->1024 self.sar_en5 = _EncoderBlock(1024, 2048, downsample=False) # 32->32 *** , 1024->2048
> > ```
> >     self.opt_en1 = _EncoderBlock(4, 64) # 256->128, 4->64
> >     self.opt_en2 = _EncoderBlock(64, 256)  # 128->64, 64->256
> >     self.opt_en3 = _EncoderBlock(256, 512)  # 64->32, 256->512
> >     self.opt_en4 = _EncoderBlock(512, 1024, downsample=False)  # 32->32 *** , 512->1024
> >     self.opt_en5 = _EncoderBlock(1024, 2048, downsample=False)  # 32->32 *** , 1024->2048
> > ```
> 
> class _EncoderBlock(nn.Module): def **init**(self, in_channels, out_channels, downsample=True): super(_EncoderBlock, self).**init**() layers = [ nn.Conv2d(in_channels, out_channels, kernel_size=3, padding=1), nn.BatchNorm2d(out_channels), nn.ReLU(inplace=True), ] if downsample: layers.append(nn.MaxPool2d(kernel_size=2, stride=2))
> 
> ```
>     self.encode = nn.Sequential(*layers)
> 
> def forward(self, x):
>     return self.encode(x)
> ```
> 
> 这个 _EncoderBlock确实只包含一个conv-BN-ReLU块

那我们最终的跑出来的结果是基于一层卷积的编码器，你可以将编码器部分更改为resnet101尝试一下，不过要注意过拟合。
